File: Go/Recursions/removeAdjDuplicate.go
package main

func removeDuplicate(str string) string {
	if len(str) == 0 {
		return ""
	}

	if len(str) == 1 {
		return str
	}

	if str[0] == str[1] {
		return removeDuplicate(str[1:])
	}

	return string(str[0]) + removeDuplicate(str[1:])
}

File: Nim/Recursions/RemoveAdjDuplicate.nim
proc remove_duplicate(str: string): string = 
    if str.len == 0:
        return ""

    if str.len == 1:
        return str

    if str[0] == str[1]:
        return remove_duplicate(str[1 ..< str.len])

    return str[0] & remove_duplicate(str[1 ..< str.len])

when isMainModule:
    let str: string = "heeelloe Worldd"

    echo remove_duplicate(str)